handlers/handlers.go

```diff
@@ -232,10 +239,41 @@ func (s *ServiceSet) serviceInfo(id string, db *mbtiles.DB) handlerFunc {
 	}
 }
 
-func (s *ServiceSet) serviceHTML(db *mbtiles.DB) handlerFunc {
+// executeTemplates first tries to find the template with the given name for
+// the ServiceSet. If that fails, it tries to instantiate it from the assets.
+// If a valid template is obtained it is used to render a response, otherwise
+// the HTTP status Internal Server Error is returned.
+func (s *ServiceSet) executeTemplate(w http.ResponseWriter, name string, data interface{}) (int, error) {
+	t := s.templates.Lookup(name)
+	var err error
+	if t == nil {
+		t, err = tmplFromAssets(s.templates, name)
+		if err != nil {
+			err = fmt.Errorf("could not parse template asset %q: %v", name, err)
+			return http.StatusInternalServerError, err
+		}
+	}
+	err = t.Execute(w, data)
+	return 0, err
```

Comment:
To prevent confusion downstream (e.g., in `wrapGetWithErrors`) where we are checking for specific response codes from the handler, it might be good to return `http.StatusOK`.  Looks like you can also return this more directly and avoid declaring `err` above.

```
return http.StatusOK, t.Execute(w, data)
```

Reply:
Done (here and on other occurrences) but I also inserted a write into a bytes.Buffer first to check whether the template could actually execute.